extension/utils.js

```diff
@@ -0,0 +1,19 @@
+window.utils = {
+	debounce(func, wait, immediate) {
+		let timeout;
+		return function (...args) {
+			const later = () => {
+				timeout = null;
+				if (!immediate) {
+					func.apply(this, args);
+				}
+			};
+			const callNow = immediate && !timeout;
+			clearTimeout(timeout);
+			timeout = setTimeout(later, wait);
+			if (callNow) {
+				func.apply(this, args);
+			}
+		};
+	}
+};
```

Comment:
I think we should use the following pattern for exporting stuff:

```js
(exports => {
  'use strict';

  exports.debounce = () => { ... }
})(window.utils = {});
```

The makes it more Node.js like.